packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -765,3 +782,230 @@ class _RenderScrollSemantics extends RenderProxyBox {
+/// respond to the different standard key bindings that cause scrolling, which
+/// are bound to keys that people use to indicate a "line" scroll (e.g.
+/// control-arrowDown keys) or a "page" scroll (e.g. pageDown key). It is
+/// recommended that at least the relative magnitudes of the scrolls match
+/// expectations.
+/// {@endtemplate}
+enum ScrollIncrementType {
+  /// Indicates that the [ScrollIncrementCalculator] should return the scroll
+  /// distance it should move when the user requests to scroll by a "line".
+  ///
+  /// The distance a "line" scrolls is really refers to what should happen when
+  /// the key binding for "scroll down/up by a line" is triggered. It's up to
+  /// the [ScrollIncrementCalculator] function to decide what that means for a
+  /// particular scrollable.
+  line,
+
+  /// Indicates that the [ScrollIncrementCalculator] should return the scroll
+  /// distance it should move when the user requests to scroll by a "page".
+  ///
+  /// The distance a "page" scrolls is really refers to what should happen when
+  /// the key binding for "scroll down/up by a page" is triggered. It's up to
+  /// the [ScrollIncrementCalculator] function to decide what that means for a
+  /// particular scrollable.
+  page,
+}
+
+/// A details object that describes the type of scroll increment being requested
+/// of a [ScrollIncrementCalculator] function, as well as the current metrics
+/// for the scrollable.
+class ScrollIncrementDetails {
+  /// A const constructor for a [ScrollIncrementDetails].
+  ///
+  /// All of the arguments must not be null, and are required.
+  const ScrollIncrementDetails({
+    @required this.type,
+    @required this.metrics,
+  })  : assert(type != null),
+        assert(metrics != null);
+
+  /// The type of scroll this is (e.g. line, page, etc.).
+  ///
+  /// {@macro flutter.widgets.scrollable.scroll_increment_type.intent}
+  final ScrollIncrementType type;
+
+  /// The current metrics of the scrollable that is being scrolled.
+  final ScrollMetrics metrics;
+}
+
+/// An [Intent] that represents scrolling the nearest scrollable by an amount
+/// appropriate for the [type] specified.
+///
+/// The actual amount of the scroll is determined by the
+/// [Scrollable.incrementCalculator], or by its defaults if that is not
+/// specified.
+class ScrollIntent extends Intent {
+  /// Creates a const [ScrollIntent] that requests scrolling in the given
+  /// [axis], with the given [type]. If [reversed] is specified, then the scroll
+  /// will happen in the opposite direction from the normal scroll direction.
+  const ScrollIntent({
+    @required this.axis,
+    this.reversed = false,
+    this.type = ScrollIncrementType.line,
+  })  : assert(axis != null),
+        assert(type != null),
+        super(ScrollAction.key);
+
+  /// The direction in which to scroll the scrollable containing the focused
+  /// widget.
+  final Axis axis;
```

Comment:
Don't you need an AxisDirection? e.g. wether to scroll up or down?

Reply:
Oh, is that supposed to be indicated by the reversed property?

Reply:
I switched back to `AxisDirection` and made sure it worked.  I never did discover what made me switch away from it in the first place: maybe an old design was flawed?  I also did add a test for RTL locales to make sure that the result was correct there too.